platform/macos/app/MapDocument.m

```diff
@@ -196,16 +197,24 @@ - (IBAction)takeSnapshot:(id)sender {
                         
                     }
                     
-                    NSString *extension = [[fileURL pathExtension] lowercaseString];
-                    NSBitmapImageFileType fileType;
-                    if ([extension isEqualToString:@"png"]) {
+                    NSString *uti;
+                    [fileURL getResourceValue:&uti forKey:NSURLTypeIdentifierKey error:nil];
```

Comment:
Turns out this doesn’t work because no file has been laid down by this point. This code ends up producing a TIFF no matter what extension is specified.

Reply:
Fixed in #10090.